### PR TITLE
fix(templates): E2E Cookiecutter - Cover all REST authentication cases + one GraphQL case

### PR DESCRIPTION
**Pull request by flexponsive**
Follow-up to #1410 

- Expands the coverage of the end-to-test for cookiecutters to cover all authentication methods for REST
- Fixes misc issues discovered by mypy, including a missing import
- Add a test case for GraphQL

----
:books: Documentation preview :books:: https://meltano-sdk--1428.org.readthedocs.build/en/1428/

### CODE REVIEW

**Comment by tayloramurphy** [comment]
Thanks for the PR @flexponsive ! We'll get one of the engineers to review @meltano/engineering

File: .github/workflows/cookiecutter-e2e.yml [hunk 1]
```diff
@@ -21,8 +24,13 @@ jobs:
       fail-fast: true
       matrix:
         include:
+          - { cookiecutter: "tap-template", replay: "tap-graphql-jwt.json", python-version: "3.9",  os: "ubuntu-latest" }
           - { cookiecutter: "tap-template", replay: "tap-rest-api_key-github.json", python-version: "3.9",  os: "ubuntu-latest" }
-          - { cookiecutter: "tap-template", replay: "tap-rest-jwt-github.json", python-version: "3.9",  os: "ubuntu-latest" }
```
Comment (edgarrmondragon) [inline .github/workflows/cookiecutter-e2e.yml:25]: You might wanna delete the corresponding replay file too

Comment (flexponsive) [reply .github/workflows/cookiecutter-e2e.yml:25]: diff display is a bit confusing here - the file was renamed `tap-rest-jwt.json` and is included three lines down

Comment (edgarrmondragon) [reply .github/workflows/cookiecutter-e2e.yml:25]: Weird, that's the intent I imagined, but I'm seeing the file when I checkout your branch:

```console
$ tree e2e-tests/cookiecutters
e2e-tests/cookiecutters
├── README.md
├── tap-graphql-jwt.json
├── tap-rest-api_key-github.json
├── tap-rest-basic_auth.json
├── tap-rest-bearer_token.json
├── tap-rest-custom.json
├── tap-rest-jwt-github.json
├── tap-rest-jwt.json
├── tap-rest-oauth2.json
├── target-per_record.json
└── test_cookiecutter.sh

1 directory, 11 files
```

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**